include/fuzzy/fuzzy_match.hxx

```diff
@@ -6,9 +6,10 @@ namespace fuzzy
   template<class Archive>
   void FuzzyMatch::save(Archive& archive, unsigned int) const
   {
+    const SuffixArrayIndex* suffixArrayIndex = _suffixArrayIndex.get();
```

Comment:
Maybe add a comment about `Fix compilation with Boost 1.58` to make sure that this won't be changed.